src/librustdoc/test.rs

```diff
@@ -308,49 +228,62 @@ fn run_test(
     };
     let output_file = outdir.path().join("rust_out");
 
-    let config = interface::Config {
-        opts: sessopts,
-        crate_cfg: config::parse_cfgspecs(cfgs),
-        input,
-        input_path: None,
-        output_file: Some(output_file.clone()),
-        output_dir: None,
-        file_loader: None,
-        diagnostic_output: DiagnosticOutput::Raw(box Sink(data.clone())),
-        stderr: Some(data.clone()),
-        crate_name: None,
-        lint_caps: Default::default(),
-    };
+    let current_dir = env::current_dir().expect("rustdoc needs to create a tempfile");
+    let input_dir = options.input.parent().unwrap_or(&current_dir);
+    let input_file = TempFileBuilder::new()
+                      .prefix("tempRustDocTestInput")
+                      .suffix(".rs")
+                      .tempfile_in(input_dir)
+                      .expect("rustdoc needs to create a tempfile");
+    std::fs::write(input_file.path(), test).expect("could write out test sources");
+
+    let mut compiler = Command::new(std::env::current_exe().unwrap().with_file_name("rustc"));
+    for cfg in &options.cfgs {
+        compiler.arg("--cfg").arg(&cfg);
+    }
+    if let Some(sysroot) = options.maybe_sysroot {
+        compiler.arg("--sysroot").arg(sysroot);
+    }
+    compiler.arg("--edition").arg(&edition.to_string());
+    compiler.arg(input_file.path()).current_dir(outdir.path());
+    compiler.arg("-o").arg(&output_file);
+    if as_test_harness {
+        compiler.arg("--test");
+    }
+    for lib_str in &options.lib_strs {
+        compiler.arg("-L").arg(&lib_str);
+    }
+    for extern_str in &options.extern_strs {
+        compiler.arg("--extern").arg(&extern_str);
+    }
+    for codegen_options_str in &options.codegen_options_strs {
+        compiler.arg("-C").arg(&codegen_options_str);
+    }
+    if let Some(linker) = options.linker {
+        compiler.arg(&format!("-C linker={:?}", linker));
```

Comment:
created https://github.com/rust-lang/rust/issues/63816 to remove this as there is already a codegen_option that handle this